gobblin-temporal/src/test/java/org/apache/gobblin/temporal/ddm/workflow/impl/AbstractNestingExecWorkflowImplTest.java

```diff
@@ -0,0 +1,139 @@
+import io.temporal.workflow.Promise;
+import io.temporal.workflow.Workflow;
+
+import org.apache.gobblin.temporal.util.nesting.work.WorkflowAddr;
+import org.apache.gobblin.temporal.util.nesting.work.Workload;
+import org.apache.gobblin.temporal.util.nesting.workflow.AbstractNestingExecWorkflowImpl;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Workflow.class)
+public class AbstractNestingExecWorkflowImplTest {
+
+  @Mock
+  private Workload<String> mockWorkload;
+
+  @Mock
+  private WorkflowAddr mockWorkflowAddr;
+
+  @Mock
+  private Workload.WorkSpan<String> mockWorkSpan;
+
+  @Mock
+  private Promise<Object> mockPromise;
+
+  private AbstractNestingExecWorkflowImpl<String, Object> workflow;
+
+  @BeforeClass
+  public void setup() {
+    // PowerMockito is required to mock static methods in the Workflow class
+    Mockito.mockStatic(Workflow.class);
+    Mockito.mockStatic(Async.class);
+    Mockito.mockStatic(Promise.class);
+    this.mockWorkload = Mockito.mock(Workload.class);
+    this.mockWorkflowAddr = Mockito.mock(WorkflowAddr.class);
+    this.mockWorkSpan = Mockito.mock(Workload.WorkSpan.class);
+    this.mockPromise = Mockito.mock(Promise.class);
+
+    workflow = new AbstractNestingExecWorkflowImpl<String, Object>() {
+      @Override
+      protected Promise<Object> launchAsyncActivity(String task) {
+        return mockPromise;
+      }
+    };
+  }
+
+  @Test
+  public void testPerformWorkload_NoWorkSpan() {
+    // Arrange
+    Mockito.when(mockWorkload.getSpan(Mockito.anyInt(), Mockito.anyInt())).thenReturn(Optional.empty());
+
+    // Act
+    int result = workflow.performWorkload(mockWorkflowAddr, mockWorkload, 0, 10, 5, Optional.empty());
+
+    // Assert
+    Assert.assertEquals(0, result);
+    Mockito.verify(mockWorkload, Mockito.times(2)).getSpan(0, 5);
+  }
+
+  @Test
+  public void testCalcPauseDurationBeforeCreatingSubTree_NoPause() {
+    // Act
+    Duration result = workflow.calcPauseDurationBeforeCreatingSubTree(50);
+
+    // Assert
+    Assert.assertEquals(Duration.ZERO, result);
+  }
+
+  @Test
+  public void testCalcPauseDurationBeforeCreatingSubTree_PauseRequired() {
+    // Act
+    Duration result = workflow.calcPauseDurationBeforeCreatingSubTree(150);
+
+    // Assert
+    Assert.assertEquals(
+        Duration.ofSeconds(AbstractNestingExecWorkflowImpl.NUM_SECONDS_TO_PAUSE_BEFORE_CREATING_SUB_TREE_DEFAULT),
+        result);
+  }
+
+  @Test
+  public void testConsolidateSubTreeGrandChildren() {
+    // Act
+    List<Integer> result = AbstractNestingExecWorkflowImpl.consolidateSubTreeGrandChildren(3, 10, 2);
+
+    // Assert
+    Assert.assertEquals(3, result.size());
+    Assert.assertEquals(Integer.valueOf(0), result.get(0));
+    Assert.assertEquals(Integer.valueOf(0), result.get(1));
+    Assert.assertEquals(Integer.valueOf(6), result.get(2));
+  }
+
+  @Test(expectedExceptions = AssertionError.class)
+  public void testPerformWorkload_LaunchesChildWorkflows() {
```

Comment:
I see you've covered some scenarios, like handling an empty WorkSpan and testing the pause duration. However, the current tests don't fully capture the functionality(creating child workflows, subtrees handling, result calculation and edge cases) of the performWorkload method. It would be useful to add overall coverage for a method when we are adding tests for it, this ensures that the tests fulfill the original intent of thoroughly validating the behavior of missing tests. This would also help avoid the need for additional work on this method again later, unless we are making changes to it.

Reply:
added temporal docker cluster to resolve this, other wise a lot of mocking would be make the tests redundant